### DISCUSSION TITLE
How to inline tailwind css styles during SSR for pages that contain getServerSideProps

### DISCUSSION THREAD

**Discussion by sambhav-gore** [vercel/next.js#22595] [Help]
In the previous versions, I was using this followin trick to inline tailwindcss styles and it was working fine before as far as I remember:

```js
// _document.js file 

import NextDocument, { Html, Head, Main, NextScript } from "next/document";
import * as fs from "fs";
import * as path from "path";

class InlineStylesHead extends Head {
  getCssLinks({ allFiles }) {
    return allFiles
      .filter((file) => file.endsWith(".css"))
      .map((file) => (
        <style
          key={file}
          nonce={this.props.nonce}
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{
            __html: fs.readFileSync(path.join(".next", file), "utf-8"),
          }}
        />
      ));
  }
}

export default class Document extends NextDocument {
  static async getInitialProps(ctx) {
    const initialProps = await NextDocument.getInitialProps(ctx);
    return { ...initialProps };
  }

  render() {
    return (
      <Html lang="en">
        <InlineStylesHead />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
```
This would read the css file and put it inline during SSR. 

but in the latest NextJS version this is not working anymore, moreover keeping this in the document is now completely breaking the styles for pages that have getServerSideProps. the error I get is that css file is not found on the path where I am trying to read it.

Has anybody any idea on how to inline the tailwindcss styles during SSR for pages that contain getServerSideProps (Lamda pages). 

FYI, it works fine for static pages but completely breaks the server side pages which is not ideal.

oh and by the way, not having the styles inline causes the page to load without styles in first render which is also not ideal. 

FYI, the rest of my setup is as following:

```js
// in _app.js

import "../styles/globals.css";

```

```css
// styles/globals.css

@tailwind base;
@tailwind components;
@tailwind utilities;
```

```js
// postcss.config.js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
```

```js
// tailwind.config.js
module.exports = {
  purge: ["./pages/**/*.js", "./components/**/*.js"],
  darkMode: false, // or 'media' or 'class'
  theme: {},
  variants: {
    extend: {},
  },
  plugins: [],
};
```

EDIT: I think it is worth mentioning that I if I build and start production build locally it does work fine, it is only in Vercel that it breaks. For some reason not able to read the static files when on vercel.

any help is highly appreciated.

**Comment by BrianHung** [comment]
I encountered the same problem with deploying critical tailwind css on vercel, and I think I found a temporary workaround? Try it out:
https://github.com/vercel/vercel/discussions/5973#discussioncomment-472618

**Comment by azizcoban** [comment]
@sambhav-gore  Hi I am just trying to do the same thing as you do. I couldn't put my entire tailwind css as inline to the head. Could you fgure out how to do that?

**Comment by Miindaugas** [comment]
Same....
